tests/lib/prepare.sh

```diff
@@ -850,17 +850,67 @@ EOF
         snap download --basename=pc-kernel --channel="20/$KERNEL_CHANNEL" pc-kernel
         # make sure we have the snap
         test -e pc-kernel.snap
         # build the initramfs with our snapd assets into the kernel snap
         uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$IMAGE_HOME"
-        EXTRA_FUNDAMENTAL="--extra-snaps $IMAGE_HOME/pc-kernel_*.snap"
+        EXTRA_FUNDAMENTAL="--snap $IMAGE_HOME/pc-kernel_*.snap"
+
+        # also add debug command line parameters to the kernel command line via
+        # the gadget in case things go side ways and we need to debug
+        snap download --basename=pc --channel="20/$GADGET_CHANNEL" pc
+        test -e pc.snap
+        unsquashfs -d pc-gadget pc.snap
+        
+        # TODO: it would be desirable when we need to do in-depth debugging of
+        # UC20 runs in google to have snapd.debug=1 always on the kernel command
+        # line, but we can't do this universally because the logic for the env
+        # variable SNAPD_DEBUG=0|false does not overwrite the turning on of 
+        # debug messages in some places when the kernel command line is set, so
+        # we get failing tests since there is extra stuff on stderr than 
+        # expected in the test when SNAPD_DEBUG is turned off
+        # so for now, don't include snapd.debug=1, but eventually it would be
+        # nice to have this on
+
+        if [ "$SPREAD_BACKEND" = "google" ]; then
+            # the default console settings for snapd aren't super useful in GCE,
+            # instead it's more useful to have all console go to ttyS0 which we 
+            # can read more easily than tty1 for example
+            for cmd in "console=ttyS0" "dangerous" "systemd.journald.forward_to_console=1" "rd.systemd.journald.forward_to_console=1" "panic=-1"; do
+                echo "$cmd" >> pc-gadget/cmdline.full
+            done 
+            # echo "snapd.debug=1" >> pc-gadget/cmdline.full
+        else
+            # but for other backends, just add the additional debugging things
+            # on top of whatever the gadget currently is configured to use
+            for cmd in "dangerous" "systemd.journald.forward_to_console=1" "rd.systemd.journald.forward_to_console=1"; do
+                echo "$cmd" >> pc-gadget/cmdline.extra
+            done 
+            # echo "snapd.debug=1" >> pc-gadget/cmdline.extra
+        fi
+
+        # TODO: this probably means it's time to move this helper out of 
+        # nested.sh to somewhere more general
+        
+        #shellcheck source=tests/lib/nested.sh
+        . "$TESTSLIB/nested.sh"
+        KEY_NAME=$(nested_get_snakeoil_key)
+
+        SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+        SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+        nested_secboot_sign_gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+        snap pack --filename=pc-repacked.snap pc-gadget 
+        mv pc-repacked.snap $IMAGE_HOME/pc-repacked.snap
```

Comment:
you might want to remove `pc-gadget` here.

Reply:
Restore will clean it up and I'd leave it in case you are in a debug shell and want to tweak/inspect it